homeassistant/components/schluter/__init__.py

```diff
@@ -0,0 +1,72 @@
+"""The Schluter DITRA-HEAT integration."""
+import logging
+
+from requests import RequestException, Session
+from schluter.api import Api
+from schluter.authenticator import AuthenticationState, Authenticator
+import voluptuous as vol
+
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+DATA_SCHLUTER_SESSION = "schluter_session"
+DATA_SCHLUTER_API = "schluter_api"
+SCHLUTER_CONFIG_FILE = ".schluter.conf"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(DOMAIN): vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up the Schluter component."""
+    _LOGGER.debug("Starting setup of schluter")
+
+    conf = config[DOMAIN]
+    api_http_session = Session()
+    api = Api(timeout=conf.get(CONF_TIMEOUT), http_session=api_http_session)
+
+    authenticator = Authenticator(
+        api,
+        conf.get(CONF_USERNAME),
+        conf.get(CONF_PASSWORD),
+        session_id_cache_file=hass.config.path(SCHLUTER_CONFIG_FILE),
```

Comment:
As a note for the future, it would be great if we don't have to pass a cache file to the authenticator and instead are able to store it in the config entry data once a config flow is added.

Reply:
Noted 👍